qemu/tests/pci_bridge.py

```diff
@@ -0,0 +1,284 @@
+        disk_num = len(disks)
+        if disk_num < image_num:
+            err_msg = "set disk num: %d" % image_num
+            err_msg += ", get in guest: %d" % disk_num
+            test.fail("Fail to list all the volumes, %s" % err_msg)
+        if not utils_disk.update_windows_disk_attributes(session, disks):
+            test.fail("Failed to update windows disk attributes.")
+        for disk in disks:
+            drive_letter = utils_disk.configure_empty_disk(session, disk,
+                                                           data_image_size,
+                                                           os_type)
+            if not drive_letter:
+                test.fail("Fail to format disks.")
+            iozone_cmd_disk = iozone_cmd % drive_letter[0]
+            status, output = session.cmd_status_output(iozone_cmd_disk,
+                                                       timeout=3600)
+            if status:
+                test.fail("Check block device '%s' failed! Output: %s"
+                          % (drive_letter[0], output))
+            utils_disk.clean_partition(session, disk, os_type)
+    else:
+        test.cancel("Unsupported OS type '%s'" % os_type)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    [pci-bridge] Check devices which attached to 1 pci-bridge
+    [pci-bridge] Check devices which attached to different pci-bridges
+    [pci-bridge] Hotplug and unplug different devices to 1 pci-bridge
+    [pci-bridge] Hotplug and unplug different devices to different pci-bridges
+    [pci-bridge] Migration with pci-bridge
+    This case will:
+    1) Attach one or multiple pci-bridge(s) to guest.
+    2) Prepare devices attached to pci-bridge(s).
+    3) Start the guest.
+    4) Hotplug devices (if needed).
+    5) Check devices.
+    6) Hotunplug devices (if needed).
+    7) Do migration (if needed).
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    error_context.context("Modify params!", logging.info)
+    pci_bridge_num = int(params.get("pci_bridge_num", 1))
+    prepare_pci_bridge(test, params, pci_bridge_num)
+    pci_bridges = params.objects("pci_controllers")
+
+    image_num = int(params.get("image_num", 3))
+    device_num = prepare_images(test, params, image_num, pci_bridge_num)
+
+    if params.get("need_hotplug_unplug", "no") == "no":
+        params["nics"] = "nic0"
+        params["nic_model_nic0"] = "virtio"
+        params["device_id_nic0"] = "nic0"
+        if pci_bridge_num == 1:
+            if device_num >= 31:
+                test.fail("There are already %d devices on %s"
+                          % (device_num, pci_bridges[0]))
+            params["nic_pci_bus_nic0"] = pci_bridges[0]
+        else:
+            index = device_num % pci_bridge_num
+            params["nic_pci_bus_nic0"] = pci_bridges[index]
+            device_num += 1
```

Comment:
L218 ~L230 is used to add one virtio-net device to pci-bridge, right ?  Why not move it to cfg,  only set `nic_pci_bus_nic0` here.